### PR TITLE
[fix](Nereids)statistics calculator for Project and Aggregate lost some columns

### PR DESCRIPTION
**Pull request by morrySnow**
## Problem summary

There are some bugs in Nereids' StatsCalculator.
- Project: return child column stats directly, so its parents cannot find column stats from project's slot.
- Aggregate: do not return column that is `Alias`, its parents cannot find some column stats from Aggregate's slot.
- All: use `SlotReference` as key of column to stats map. So we need change SlotReference's `equals` and `hashCode` method to just using `ExprId` as we discussed.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 1]
```diff
@@ -250,46 +253,53 @@ private StatsDeriveResult computeLimit(Limit limit) {
...
+        Map<Slot, ColumnStats> childColumnStats = groupExpression.getCopyOfChildStats(0).getSlotToColumnStats();
+        StatsDeriveResult statsDeriveResult = new StatsDeriveResult(groupExpression.getCopyOfChildStats(0));
+        Map<Slot, ColumnStats> columnsStats = projections.stream().map(projection -> {
+            List<SlotReference> slotReferences = projection.collect(SlotReference.class::isInstance);
+            if (slotReferences.isEmpty()) {
+                ColumnStats columnStats = new ColumnStats();
+                columnStats.setAvgSize(1);
+                columnStats.setMaxSize(1);
+                columnStats.setNdv(1);
+                columnStats.setNumNulls(0);
```
Comment (jackwener) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:296]: we can add `newDefaultColumnStats()`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 2]
```diff
@@ -250,46 +253,53 @@ private StatsDeriveResult computeLimit(Limit limit) {
...
-            List<Slot> slotReferenceList = namedExpression.collect(SlotReference.class::isInstance);
-            return slotReferenceList.stream();
-        }).collect(Collectors.toList());
-        StatsDeriveResult stat = groupExpression.getCopyOfChildStats(0);
-        Map<Slot, ColumnStats> slotColumnStatsMap = stat.getSlotToColumnStats();
-        slotColumnStatsMap.entrySet().removeIf(entry -> !slotSet.contains(entry.getKey()));
-        return stat;
+        List<NamedExpression> projections = project.getProjects();
+        Map<Slot, ColumnStats> childColumnStats = groupExpression.getCopyOfChildStats(0).getSlotToColumnStats();
+        StatsDeriveResult statsDeriveResult = new StatsDeriveResult(groupExpression.getCopyOfChildStats(0));
```
Comment (Kikyou1997) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:288]: Why new stats here?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 3]
```diff
@@ -250,46 +253,53 @@ private StatsDeriveResult computeLimit(Limit limit) {
...
+            List<SlotReference> slotReferences = projection.collect(SlotReference.class::isInstance);
+            if (slotReferences.isEmpty()) {
+                ColumnStats columnStats = new ColumnStats();
+                columnStats.setAvgSize(1);
+                columnStats.setMaxSize(1);
+                columnStats.setNdv(1);
+                columnStats.setNumNulls(0);
+                return new AbstractMap.SimpleEntry<>(projection.toSlot(), columnStats);
+            } else {
+                return new AbstractMap.SimpleEntry<>(projection.toSlot(), childColumnStats.get(slotReferences.get(0)));
```
Comment (Kikyou1997) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:299]: Why take first slot from slotReferences list

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:295]: just a trick, need fix later

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java [hunk 4]
```diff
@@ -250,46 +253,49 @@ private StatsDeriveResult computeLimit(Limit limit) {
...
-            List<Slot> slotReferenceList = namedExpression.collect(SlotReference.class::isInstance);
-            return slotReferenceList.stream();
-        }).collect(Collectors.toList());
-        StatsDeriveResult stat = groupExpression.getCopyOfChildStats(0);
-        Map<Slot, ColumnStats> slotColumnStatsMap = stat.getSlotToColumnStats();
-        slotColumnStatsMap.entrySet().removeIf(entry -> !slotSet.contains(entry.getKey()));
-        return stat;
+        List<NamedExpression> projections = project.getProjects();
+        Map<Slot, ColumnStats> childColumnStats = groupExpression.getCopyOfChildStats(0).getSlotToColumnStats();
+        StatsDeriveResult statsDeriveResult = groupExpression.getCopyOfChildStats(0);
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java:288]:
```suggestion
        StatsDeriveResult statsDeriveResult = groupExpression.getCopyOfChildStats(0);
        Map<Slot, ColumnStats> childColumnStats = statsDeriveResult.getSlotToColumnStats();
```

**Review by 924060929 — DISMISSED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**